pytest_inmanta/module/init.py

```diff
@@ -54,6 +62,9 @@ def read_resource(
         if resource.fail:
             raise handler.InvalidOperation()
 
+        if resource.fail_deploy and not ctx.is_dry_run():
```

Comment:
In addition to this there is also the check `if resource.name in DATA` above. I'm not sure if we need to take that in to account as well. @wouterdb wdyt?

Reply:
I'm approving the PR but I would await Wouter's response before merging. The same question applies to the `wrong_diff`.

Reply:
ok, this does aan new functionality, so I retract my comment